### ISSUE TITLE
--fix for prefer-interfaces erases generics for functions (in 1.1.3)

### ISSUE THREAD

**Issue by grumd** [cartant/eslint-plugin-etc#23]
Sorry, I forgot to include this in my previous bug report.

Input:

```ts
type MyFunction<Foo> = <Bar>(foo: Foo) => Bar;
```

Expected output:

```ts
interface MyFunction<Foo> { <Bar>(foo: Foo): Bar; }
```

Actual output:

```ts
interface MyFunction<Foo> { (foo: Foo): Bar; }
```

**Comment by cartant** [comment]
Yeah, that should be easy to fix, too.

**Comment by cartant** [comment]
Should be fixed in 1.1.4.
